### ISSUE TITLE
[Bug] T.tile.broadcast pto 后端 1D 标量广播结果错误（TCOLEXPAND 读取不完整 src tile）

### ISSUE THREAD

**Issue by 2Sheldonlaw** [tile-ai/tilelang-ascend#1695]
## Bug Description

`T.tile.broadcast` 在 pto 后端对 1D src `(1,)` → dst `(N,)` axis=0 做标量广播时，只有第一个元素正确，其余全为 0。

根因在 pto codegen 的数据加载和 `TCOLEXPAND` 指令之间的不匹配：

- `copy_gm_to_ub_dynamic` 只加载了 1 个有效元素到 `src_ub`（`validRow=1, validCol=1`），但 pto tile 最小列宽是 8，`src_ub` 的物理 tile 是 `1×8`，其中只有 1 个有效元素，其余 7 个未初始化
- `TCOLEXPAND(dst_ub, src_ub)` 读取 `src_ub` 的完整 tile 行来广播，但只有 1 个有效元素，广播结果只有第一个元素正确

ascendc 后端不受影响——ascendc codegen 用 `AscendC::Broadcast` API，内部自行处理标量广播。1D→2D auto-infer（axis=None）在 pto 上也正常——走 2D 路径，src 有完整数据行。

**仓库 examples 不触发此问题**——没有 example 使用 1D pto broadcast。

## Environment

- NPU: Ascend910_9362 (A3, dav-c220)
- CPU Architecture: aarch64
- CANN: 9.0.0
- torch: 2.7.1+cpu
- torch-npu: 2.7.1.post4
- Python: 3.11.4
- TileLang-Ascend: `5149acdb40b086d7cd84e30821f5759fa8699a14`
- Installation: build from source

## Code to Reproduce

```bash
source set_env.sh
python repro_broadcast_bug4.py
```

<details>
<summary>完整复现代码</summary>

```python
import torch
import tilelang
import tilelang.language as T

tilelang.cache.clear_cache()

pass_configs = {
    tilelang.PassConfigKey.TL_ASCEND_AUTO_CV_COMBINE: True,
    tilelang.PassConfigKey.TL_ASCEND_AUTO_CV_SYNC: True,
    tilelang.PassConfigKey.TL_ASCEND_AUTO_SYNC: True,
    tilelang.PassConfigKey.TL_ASCEND_MEMORY_PLANNING: True,
}

N = 64


@T.prim_func
def broadcast_1d(A: T.Tensor((1,), "float32"), B: T.Tensor((N,), "float32")):
    with T.Kernel(1, is_npu=True) as (cid, vid):
        src_ub = T.alloc_ub((1,), "float32")
        dst_ub = T.alloc_ub((N,), "float32")
        T.copy(A, src_ub)
        T.tile.broadcast(dst_ub, src_ub, axis=0)
        T.copy(dst_ub, B)


val = 42.0
ref = torch.full((N,), val, dtype=torch.float32)

print("=== 1D broadcast pto ===")

# pto (bug)
kernel = tilelang.compile(broadcast_1d, out_idx=[-1], pass_configs=pass_configs, target="pto")
a = torch.tensor([val], dtype=torch.float32).npu()
b = kernel(a)
torch.npu.synchronize()
match = torch.equal(b.cpu(), ref)
print(f"pto: {'PASS' if match else 'FAIL'} (max_diff={(b.cpu()-ref).abs().max().item()})")
if not match:
    print(f"  b[:8]: {b.cpu()[:8].tolist()}")

# ascendc (control)
kernel = tilelang.compile(broadcast_1d, out_idx=[-1], pass_configs=pass_configs, target="ascendc")
b = kernel(a)
torch.npu.synchronize()
match = torch.equal(b.cpu(), ref)
print(f"ascendc: {'PASS' if match else 'FAIL'}")
```

</details>

### Reproduction results

```
pto: FAIL (max_diff=125.12718963623047)
  b[:8]: [42.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
ascendc: PASS
```

只有第一个元素 42.0 正确，其余 63 个全为 0。

## Error Message

无编译错误、无运行时错误——**静默失败**。编译通过，运行通过，但结果错误。

## Expected Behavior

`T.tile.broadcast` 应在 pto 后端正确处理 1D 标量广播（src=(1,) → dst=(N,)）。

## Actual Behavior

pto codegen 生成的代码：

```cpp
// 1D broadcast (FAIL):
// copy_gm_to_ub_dynamic 只加载 1 个元素到 src_ub (validRow=1, validCol=1)
copy_gm_to_ub_dynamic<float, ..., 1, 8, ...>(..., 0, 0, 1, 1);  // validRow=1, validCol=1
TCOLEXPAND(dst_ub, src_ub);  // src_ub 物理 tile 是 1×8，但只有 1 个有效元素

// 2D broadcast (PASS):
// copy_gm_to_ub_dynamic 加载完整数据行到 src_ub (validRow=1, validCol=64)
copy_gm_to_ub_dynamic<float, ..., 1, 64, ...>(..., 0, 0, 1, 64);  // validRow=1, validCol=64
TCOLEXPAND(dst_ub, src_ub);  // src_ub 有完整数据行，广播正确
```

1D 和 2D 都用 `TCOLEXPAND`，但 1D 时 src_ub 只有 1 个有效元素（物理 tile 是 `1×8`，validCol=1），`TCOLEXPAND` 读取不完整的数据行导致广播结果错误。

### 方案 A：在 pto codegen 中检测 1D 标量广播

在 `CodegenColBroadcast` 中检测 src 是 1D 且只有 1 个元素时，改用 V-pipe 指令（如 `Duplicate`）填充 dst_ub，而非 `TCOLEXPAND`。

### 方案 B：在前端把 1D broadcast 自动转为 2D

在 `ascend_tile.py` 的 `broadcast` 函数中，检测 src 是 1D 且 src_extent[0]==1 时，自动转为 2D 处理（和 auto-infer 一致）。1D→2D auto-infer 在 pto 上已验证通过。

### 方案 C：修复 copy_gm_to_ub_dynamic 的数据加载

让 `copy_gm_to_ub_dynamic` 在加载标量数据时用 `Duplicate` 填充整个 tile 行，确保 `TCOLEXPAND` 能读到完整数据。

## Additional Context

**Backend:**
- [ ] Ascend C backend（不受影响）
- [x] PTO backend（受影响）
- [ ] AscendNPU IR backend

**不受影响**：
- ascendc 后端（用 `AscendC::Broadcast` API，内部处理标量广播）
- 1D→2D auto-infer（axis=None，走 2D 路径，src 有完整数据行）
- 2D broadcast axis=0（src=(1,N)→dst=(M,N)，src 有完整数据行）

<details>
<summary>源码调用链</summary>

### TileLang frontend

`tilelang/language/ascend_tile.py` — `broadcast()` 函数

1D src `(1,)` → dst `(N,)` axis=0 保持 dim=1，生成 `Broadcast<float, 1, 0, false>`。

### TileLang codegen (pto)

`src/target/codegen_ascend_pto.cc:2906-2919` — `BroadcastOpCodegen`

```cpp
void CodeGenTileLangAscendPto::BroadcastOpCodegen(const CallNode *op) {
    std::string axis = extractBroadCastAxis(template_args);
    if (axis == "1") {
        CodegenRowBroadcast(dst_shape_info, src_shape_info);   // TROWEXPAND
    } else {
        CodegenColBroadcast(dst_shape_info, src_shape_info);   // TCOLEXPAND
    }
}
```

`src/target/codegen_ascend_pto.cc:2781-2799` — `CodegenColBroadcast`

```cpp
void CodeGenTileLangAscendPto::CodegenColBroadcast(const ShapeInfo &dst, const ShapeInfo &src) {
    // 生成：TCOLEXPAND(dst_ub, src_ub);
}
```

### 问题所在

1D 时 `copy_gm_to_ub_dynamic` 生成的参数：

```cpp
copy_gm_to_ub_dynamic<float, float, 1, 1, 1, 1, 8, 1, 1, 1, 1, 1, 1, 8, pto::PadValue::Zero>(
    A_handle + 0, pto::Shape<1, 1, 1, 1, 8>(), pto::Stride<1, 1, 1, 1, 1>(),
    0, 0, 1, 1);  // validRow=1, validCol=1 ← 只加载 1 个元素
```

`src_ub` 的物理 tile 是 `TileUbDataND<float, 1, 8>`（1 行 8 列），但只有 1 个有效元素。`TCOLEXPAND` 读取整个 tile 行时只有第一个元素有效，广播结果只有第一个元素正确。

### Examples 规避方式

没有 example 使用 1D pto broadcast。examples 中的 broadcast 都是 2D（src=(1,N)→dst=(M,N) 或 src=(M,1)→dst=(M,N)），src 有完整数据行，不触发此 bug。

</details>
